front/components/data_source/RequestDataSourcesModal.tsx

```diff
@@ -0,0 +1,181 @@
+import { Button, DropdownMenu, Modal, TextArea } from "@dust-tt/sparkle";
+import React, { useContext, useState } from "react";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { CONNECTOR_CONFIGURATIONS } from "@app/lib/connector_providers";
+import { sendEmail } from "@app/lib/email";
+import logger from "@app/logger/logger";
+import type { DataSourceIntegration } from "@app/pages/w/[wId]/builder/data-sources/managed";
+
+type RequestDataSourceProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  dataSourceIntegrations: DataSourceIntegration[];
+  currentUserEmail: string;
+};
+
+async function sendRequestDataSourceEmail(
```

Comment:
This part should live gated behind a backend endpoint 🙏. Also we should use sendgrid with the template to send email now.